### PR TITLE
Update match_endofsentence to use NLTK sentence tokenizer

### PR DESCRIPTION
**Pull request by markbackman**
#### Please describe the changes in your PR. If it is addressing an issue, please reference that as well.

The regex based end of sentence detection is too fragile and has too many edge cases. Instead, I'm proposing that we move to using NLTK's punkt_tab sentence tokenizer along with a secondary check for punctuation.

The secondary check is required due to the LLM streaming token use case—NLTK will see an individual word, like `Hello`, and deem it as a complete sentence. Since all LLM output is streamed as tokens, this would result in TTS aggregation breaking. Instead, we're interested in NLTK's sentence detection (and edge case handling for things like acronyms, numbers, emails, etc) along with an added check that the sentence ends in some form of punctuation. Since Pipecat supports a wide range of languages, I asked Claude for some help determining end of sentence punctuation for a broader set of languages. We can expand this as needed.

I also expanded tests to include some edge cases that were failing previously.

Note: `nltk` is now a required dependency with Pipecat.

Should resolve or improve: #2040, #1692, #1994.

### CODE REVIEW

File: tests/test_utils_string.py [hunk 1]
```diff
@@ -16,52 +16,177 @@ async def test_endofsentence(self):
         assert match_endofsentence("This is a sentence?") == 19
         assert match_endofsentence("This is a sentence;") == 19
         assert match_endofsentence("This is a sentence...") == 21
-        assert match_endofsentence("This is a sentence . . .") == 24
-        assert match_endofsentence("This is a sentence. ..") == 22
+        assert match_endofsentence("This is a sentence . . .") == 20
```
Comment (filipi87) [inline tests/test_utils_string.py]: If I am not mistaken, before we were considering this as one sentence:
"This is a sentence . . ."

But now, we are considering that the end of sentence finished at the first "."

Is it expected?

Comment (markbackman) [reply tests/test_utils_string.py]: This is a weird punctuation situation, which really is more of a stretch to test the logic than it is a case we need to handle. Maybe we should just remove it since it's confusing. WDYT?

Comment (filipi87) [reply tests/test_utils_string.py]: Yeah, I think it's not something we need to handle, so it's better to remove it. 👍

Comment (markbackman) [reply tests/test_utils_string.py]: I'll replace with a better, more realistic test.
